bats/sql-reset.bats

```diff
@@ -0,0 +1,134 @@
+
+    dolt sql <<SQL
+CREATE TABLE test (
+    pk int primary key
+);
+SQL
+
+    dolt commit -a -m "Add a table"
+}
+
+teardown() {
+    teardown_common
+}
+
+@test "DOLT_RESET --hard works on unstaged and staged table changes" {
+    dolt sql -q "INSERT INTO test VALUES (1)"
+
+    run dolt sql -q "SELECT DOLT_RESET('--hard')"
+    [ $status -eq 0 ]
+
+    run dolt status
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "On branch master" ]] || false
+    [[ "$output" =~ "nothing to commit, working tree clean" ]] || false
+
+    dolt sql -q "INSERT INTO test VALUES (1)"
+
+    dolt add .
+
+    run dolt sql -q "SELECT DOLT_RESET('--hard')"
+    [ $status -eq 0 ]
+
+    run dolt status
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "On branch master" ]] || false
+    [[ "$output" =~ "nothing to commit, working tree clean" ]] || false
+}
+
+@test "DOLT_RESET --hard does not ignore staged docs" {
+    # New docs gets referred as untracked file.
+    echo ~license~ > LICENSE.md
+    dolt add .
+
+    run dolt sql -q "SELECT DOLT_RESET('--hard')"
+    [ $status -eq 0 ]
+
+    run dolt status
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "Untracked files:" ]] || false
+    [[ "$output" =~ ([[:space:]]*new doc:[[:space:]]*LICENSE.md) ]] || false
+
+    # Tracked file gets reset
+    dolt commit -a -m "Add a the license file"
+    echo ~edited-license~ > LICENSE.md
+
+    dolt add .
+
+    run dolt sql -q "SELECT DOLT_RESET('--hard')"
+    [ $status -eq 0 ]
+
+    run dolt status
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "Changes not staged for commit:" ]] || false
+    [[ "$output" =~ ([[:space:]]*modified:[[:space:]]*LICENSE.md) ]] || false
+}
+
+@test "DOLT_RESET --soft works on unstaged and staged table changes" {
+    dolt sql -q "INSERT INTO test VALUES (1)"
+
+    # Table should still be unstaged
+    run dolt sql -q "SELECT DOLT_RESET('--soft')"
+    [ $status -eq 0 ]
+
+    run dolt status
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "Changes not staged for commit:" ]] || false
+    [[ "$output" =~ ([[:space:]]*modified:[[:space:]]*test) ]] || false
+
+    dolt add .
+
+    run dolt sql -q "SELECT DOLT_RESET('--soft')"
+    [ $status -eq 0 ]
+
+    run dolt status
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "Changes not staged for commit:" ]] || false
+    [[ "$output" =~ ([[:space:]]*modified:[[:space:]]*test) ]] || false
+}
+
+@test "DOLT_RESET --soft ignores staged docs" {
+    echo ~license~ > LICENSE.md
+    dolt add .
+
+    run dolt sql -q "SELECT DOLT_RESET('--soft')"
+    [ $status -eq 0 ]
+
+    run dolt status
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "Changes to be committed:" ]] || false
+    [[ "$output" =~ ([[:space:]]*new doc:[[:space:]]*LICENSE.md) ]] || false
+
+    # Explicitly defining the file ignores it.
+    run dolt sql -q "SELECT DOLT_RESET('LICENSE.md')"
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ ("error: the table(s) LICENSE.md do not exist") ]] || false
+}
+
+@test "DOLT_RESET works on specific tables" {
```

Comment:
Should have at least one test for combining --hard or --soft with a table argument